### ISSUE TITLE
AssertionError in solveset(sinh(8*x) + coth(12*x))

### ISSUE THREAD

**Issue by gschintgen** [sympy/sympy#19489]
```
In [26]: solveset(sinh(8*x) + coth(12*x))                                                                            [4/1675]---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-26-cc67099ea05a> in <module>
----> 1 solveset(sinh(8*x) + coth(12*x))

~/sympy/sympy/solvers/solveset.py in solveset(f, symbol, domain)
   2037     f = piecewise_fold(f)
   2038
-> 2039     return _solveset(f, symbol, domain, _check=True)
   2040
   2041

~/sympy/sympy/solvers/solveset.py in _solveset(f, symbol, domain, _check)
    933     elif _is_function_class_equation(TrigonometricFunction, f, symbol) or \
    934             _is_function_class_equation(HyperbolicFunction, f, symbol):
--> 935         result = _solve_trig(f, symbol, domain)
    936     elif isinstance(f, arg):
    937         a = f.args[0]

~/sympy/sympy/solvers/solveset.py in _solve_trig(f, symbol, domain)
    542     if sol1 is None or isinstance(sol1, ConditionSet):
    543         try:
--> 544             sol = _solve_trig2(f, symbol, domain)
    545         except ValueError:
    546             sol = sol1

~/sympy/sympy/solvers/solveset.py in _solve_trig2(f, symbol, domain)
    626         mu = Rational(2)*ilcm(*denominators)/igcd(*numerators)
    627     else:
--> 628         assert len(numerators) == 1
    629         mu = Rational(2)*denominators[0]/numerators[0]
    630

AssertionError:
```
Currently `_solve_trig2` is not actually designed to handle hyperbolic functions. It's possible to extend this solver to hyperbolic functions, but for now those equations shouldn't end up there.

**Comment by gschintgen** [comment]
This is actually a regression in Sympy 1.6.
Sympy 1.5.1 returns an (unhelpful) `ConditionSet` but it doesn't raise an exception.

**Comment by gschintgen** [comment]
I added a new "regression" label in github. I hope that's ok? I also added it to the 1.6.1 milestone. (It should be fairly easy to patch up.)

**Comment by gschintgen** [comment]
I think I filed this one twice. There's also #18490.

**Comment by oscarbenjamin** [comment]
I'm going to reopen this until it's merged for 1.6.1.

**Comment by oscarbenjamin** [comment]
#19602 applies #19566 to the 1.6 release branch so that this will be fixed in 1.6.1 (see #19599)

**Comment by gschintgen** [comment]
Thanks @oscarbenjamin
